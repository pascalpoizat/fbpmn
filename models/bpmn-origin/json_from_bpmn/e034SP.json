{
    "isInterrupt": {},
    "catN": {
        "ExclusiveGateway_0vdfgj4": "XorGateway",
        "EndEvent_0a59w28": "NoneEndEvent",
        "StartEvent_1": "NoneStartEvent",
        "StartEvent_09ojxru": "NoneStartEvent",
        "P_": "Process",
        "EndEvent_0atuxkh": "NoneEndEvent",
        "StartEvent_0p1e2cq": "NoneStartEvent",
        "SubProcess_1glz8ii": "SubProcess",
        "Task_1x5zvyz": "AbstractTask"
    },
    "catE": {
        "SequenceFlow_1t5jo8c": "NormalSequenceFlow",
        "SequenceFlow_04a621f": "NormalSequenceFlow",
        "SequenceFlow_12utbfs": "NormalSequenceFlow",
        "SequenceFlow_00j4ong": "NormalSequenceFlow",
        "SequenceFlow_19k9u87": "NormalSequenceFlow",
        "SequenceFlow_1uo79pd": "NormalSequenceFlow"
    },
    "containN": {
        "P_": [
            "StartEvent_1",
            "SubProcess_1glz8ii",
            "EndEvent_0a59w28"
        ],
        "SubProcess_1glz8ii": [
            "StartEvent_09ojxru",
            "StartEvent_0p1e2cq",
            "Task_1x5zvyz",
            "EndEvent_0atuxkh",
            "ExclusiveGateway_0vdfgj4"
        ]
    },
    "messageE": {},
    "timeInformation": {},
    "targetE": {
        "SequenceFlow_1t5jo8c": "ExclusiveGateway_0vdfgj4",
        "SequenceFlow_04a621f": "Task_1x5zvyz",
        "SequenceFlow_12utbfs": "SubProcess_1glz8ii",
        "SequenceFlow_00j4ong": "EndEvent_0a59w28",
        "SequenceFlow_19k9u87": "ExclusiveGateway_0vdfgj4",
        "SequenceFlow_1uo79pd": "EndEvent_0atuxkh"
    },
    "name": "e034SP",
    "attached": {},
    "containE": {
        "P_": [
            "SequenceFlow_12utbfs",
            "SequenceFlow_00j4ong"
        ],
        "SubProcess_1glz8ii": [
            "SequenceFlow_19k9u87",
            "SequenceFlow_1t5jo8c",
            "SequenceFlow_04a621f",
            "SequenceFlow_1uo79pd"
        ]
    },
    "edges": [
        "SequenceFlow_12utbfs",
        "SequenceFlow_00j4ong",
        "SequenceFlow_19k9u87",
        "SequenceFlow_1t5jo8c",
        "SequenceFlow_04a621f",
        "SequenceFlow_1uo79pd"
    ],
    "messages": [],
    "sourceE": {
        "SequenceFlow_1t5jo8c": "StartEvent_0p1e2cq",
        "SequenceFlow_04a621f": "ExclusiveGateway_0vdfgj4",
        "SequenceFlow_12utbfs": "StartEvent_1",
        "SequenceFlow_00j4ong": "SubProcess_1glz8ii",
        "SequenceFlow_19k9u87": "StartEvent_09ojxru",
        "SequenceFlow_1uo79pd": "Task_1x5zvyz"
    },
    "nodes": [
        "P_",
        "StartEvent_1",
        "SubProcess_1glz8ii",
        "EndEvent_0a59w28",
        "StartEvent_09ojxru",
        "StartEvent_0p1e2cq",
        "Task_1x5zvyz",
        "EndEvent_0atuxkh",
        "ExclusiveGateway_0vdfgj4"
    ],
    "nameN": {
        "SubProcess_1glz8ii": "SP"
    }
}