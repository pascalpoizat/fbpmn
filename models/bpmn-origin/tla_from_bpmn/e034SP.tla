---------------- MODULE e034SP ----------------

EXTENDS TLC, PWSTypes

VARIABLES nodemarks, edgemarks, net

Interest ==
  "P_" :> {  }

ContainRel ==
  "P_" :> { "StartEvent_1", "SubProcess_1glz8ii", "EndEvent_0a59w28" }
  @@ "SubProcess_1glz8ii" :> { "StartEvent_09ojxru", "StartEvent_0p1e2cq", "Task_1x5zvyz", "EndEvent_0atuxkh", "ExclusiveGateway_0vdfgj4" }

Node == { "P_", "StartEvent_1", "SubProcess_1glz8ii", "EndEvent_0a59w28", "StartEvent_09ojxru", "StartEvent_0p1e2cq", "Task_1x5zvyz", "EndEvent_0atuxkh", "ExclusiveGateway_0vdfgj4" }

Edge == { "SequenceFlow_12utbfs", "SequenceFlow_00j4ong", "SequenceFlow_19k9u87", "SequenceFlow_1t5jo8c", "SequenceFlow_04a621f", "SequenceFlow_1uo79pd" }

Message == {  }

msgtype ==
  [ i \in {} |-> {}]


source ==
   "SequenceFlow_12utbfs" :> "StartEvent_1"
@@ "SequenceFlow_00j4ong" :> "SubProcess_1glz8ii"
@@ "SequenceFlow_19k9u87" :> "StartEvent_09ojxru"
@@ "SequenceFlow_1t5jo8c" :> "StartEvent_0p1e2cq"
@@ "SequenceFlow_04a621f" :> "ExclusiveGateway_0vdfgj4"
@@ "SequenceFlow_1uo79pd" :> "Task_1x5zvyz"

target ==
   "SequenceFlow_12utbfs" :> "SubProcess_1glz8ii"
@@ "SequenceFlow_00j4ong" :> "EndEvent_0a59w28"
@@ "SequenceFlow_19k9u87" :> "ExclusiveGateway_0vdfgj4"
@@ "SequenceFlow_1t5jo8c" :> "ExclusiveGateway_0vdfgj4"
@@ "SequenceFlow_04a621f" :> "Task_1x5zvyz"
@@ "SequenceFlow_1uo79pd" :> "EndEvent_0atuxkh"

CatN ==
   "P_" :> Process
@@ "StartEvent_1" :> NoneStartEvent
@@ "SubProcess_1glz8ii" :> SubProcess
@@ "EndEvent_0a59w28" :> NoneEndEvent
@@ "StartEvent_09ojxru" :> NoneStartEvent
@@ "StartEvent_0p1e2cq" :> NoneStartEvent
@@ "Task_1x5zvyz" :> AbstractTask
@@ "EndEvent_0atuxkh" :> NoneEndEvent
@@ "ExclusiveGateway_0vdfgj4" :> ExclusiveOr

CatE ==
   "SequenceFlow_12utbfs" :> NormalSeqFlow
@@ "SequenceFlow_00j4ong" :> NormalSeqFlow
@@ "SequenceFlow_19k9u87" :> NormalSeqFlow
@@ "SequenceFlow_1t5jo8c" :> NormalSeqFlow
@@ "SequenceFlow_04a621f" :> NormalSeqFlow
@@ "SequenceFlow_1uo79pd" :> NormalSeqFlow

LOCAL preEdges ==
  [ i \in {} |-> {}]
PreEdges(n,e) == preEdges[n,e]

PreNodes(n,e) == { target[ee] : ee \in preEdges[n,e] }
          \union { nn \in { source[ee] : ee \in preEdges[n,e] } : CatN[nn] \in { NoneStartEvent, MessageStartEvent } }

BoundaryEvent ==
  [ i \in {} |-> {}]

WF == INSTANCE PWSWellFormed
ASSUME WF!WellTyped
ASSUME WF!WellFormedness

ConstraintNode == TRUE \* none
ConstraintEdge == TRUE \* none
Constraint == TRUE     \* none
INSTANCE PWSConstraints
INSTANCE UserProperties
INSTANCE PWSSemantics

================================================================

