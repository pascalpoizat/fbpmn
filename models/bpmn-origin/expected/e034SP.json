{
  "e034SP": {
    "Network02FifoPair": {},
    "Network03Causal": {},
    "Network07RSC": {},
    "Network05Outbox": {},
    "Network04Inbox": {},
    "Network01Bag": {},
    "Network06Fifo": {}
  }
}
